src/main/flight/rpm_filter.c

```diff
@@ -112,7 +111,7 @@ void rpmFilterInit(const rpmFilterConfig_t *config, const timeUs_t looptimeUs)
 
 FAST_CODE_NOINLINE void rpmFilterUpdate(void)
 {
-    if (!motorConfig()->dev.useDshotTelemetry) {
+    if (!useDshotTelemetry || (rpmFilter.numHarmonics == 0)) {
```

Comment:
```suggestion
    if (!useDshotTelemetry) {
```
`rpmFilter.numHarmonics == 0` is handled below in `bool isRpmFilterEnabled()`

Reply:
Agreed. That was an early change I made, but I failed to revert.